### PR TITLE
feat(buttons): adding in color changes throughout applicable components

### PR DESCRIPTION
**Pull request by ccedrone**
V3 button styling updates

### CODE REVIEW

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:213]: Is the marketing button the yellow (secondary) one?

File: .changeset/afraid-coats-laugh.md
Comment (silvalaura) [inline .changeset/afraid-coats-laugh.md:5]: We should note here that `outline` and `success` buttons went away because those are breaking changes.

File: packages/react-magma-dom/src/components/Pagination/Pagination.test.js
Comment (silvalaura) [inline packages/react-magma-dom/src/components/Pagination/Pagination.test.js:51]: Should we use `transparentize` here to match the code?

File: website/react-magma-docs/src/pages/api/button.mdx
Comment (silvalaura) [inline website/react-magma-docs/src/pages/api/button.mdx:47]: The description above (line 39) can be updated to remove `success`

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:308]: Just to triple check, what is the difference between `lighten 0.2` and `transparentize 0.2`-- are they interchangeable?

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:185]: Is -0.8 === 0.2? The negative number here feels weird

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [reply packages/react-magma-dom/src/components/StyledButton/styles.ts:308]: Cool! I guess `lighten` still makes sense for the active state. Thanks

**Review by silvalaura — APPROVED** [review]

**Review by greenervigil — APPROVED** [review]

**Review by orion-cengage — CHANGES_REQUESTED** [review]
- See AC 1.2  -  font-weight should be Medium (500)
- On Secondary button Hover, the background is turning gray.  It should be doing the same thing as the Primary Link button where the background turns light purple and the text doesn't really do anything or maybe it gets a little darker.
image
image
- There are some other things we're going to tweak on the hover and pressed states but we'll handle that in a separate ticket.
- There isn't a Secondary version of the Link button since it just looks like primary, so we shouldn't show it.
image

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:141]: Can we update this to remove `neutral06`?

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:187]: These values are the same, so maybe we can remove the if?

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:155]: Let's move this isInverse check to line 143 so if it's Inverse and Secondary, it returns the correct color

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:112]: I think you can remove the `&& !props.isInverse` check because if it is inverse, it would've gone into the block above

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:92]: If it can be link && secondary, which is the expected value? Just want to make sure these two checks are in the correct order

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:140]: Can this be removed if the color is the same as the default if it's not inverse?

**Review by orion-cengage — APPROVED** [review]

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:198]: This is a duplicate of lines 208-209

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:211]: I'm probably missing it, but do we have the check for `danger` when not inverse and not disabled, or do we expect it to be this default?

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:220]: `transparentize(0.3, neutral900)`

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:226]: This check for is danger, not solid and not inverse is in two places: line 226 and line 237. I think this one can be removed and line 237 should be updated with the correct value

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [reply packages/react-magma-dom/src/components/StyledButton/styles.ts:220]: Line 242 checks also checks for `props.variant !== 'solid' && props.isInverse`. Can we move the color check there?

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:290]: Move marketing check from line 257 to this switch statement, and make sure this switch has a default.

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:261]: Let's replace this usage of `foundation02`

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:335]: This check is a dupe of line 346

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:329]: transparentize(0.3, neutral900);

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:377]: Dupe of line 388

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:373]: Let's remove the usage of this old color

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:371]: Can this marketing check be moved to the switch statement on line 394?

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:394]: This switch statement needs a default

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:406]: This should be condensed as is a duplicate of the if statement on line 393

**Review by silvalaura — CHANGES_REQUESTED** [review]
Let's try to clean up the styling branching logic a little bit more before merging.

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (ccedrone) [reply packages/react-magma-dom/src/components/StyledButton/styles.ts:406]: Done

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (ccedrone) [reply packages/react-magma-dom/src/components/StyledButton/styles.ts:371]: Done

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (ccedrone) [reply packages/react-magma-dom/src/components/StyledButton/styles.ts:373]: Done

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (ccedrone) [reply packages/react-magma-dom/src/components/StyledButton/styles.ts:353]: Done

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (ccedrone) [reply packages/react-magma-dom/src/components/StyledButton/styles.ts:261]: Done

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (ccedrone) [reply packages/react-magma-dom/src/components/StyledButton/styles.ts:290]: Done

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (ccedrone) [reply packages/react-magma-dom/src/components/StyledButton/styles.ts:220]: Done

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (ccedrone) [reply packages/react-magma-dom/src/components/StyledButton/styles.ts:329]: Done

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [inline packages/react-magma-dom/src/components/StyledButton/styles.ts:155]: nitpick: with one case, change to use if statement instead of switch

File: packages/react-magma-dom/src/components/StyledButton/styles.ts
Comment (silvalaura) [reply packages/react-magma-dom/src/components/StyledButton/styles.ts:209]: Looks like it expects to be white, so it defaults which is fine.

File: packages/react-magma-dom/src/components/Button/Button.stories.tsx
Comment (orion-cengage) [reply packages/react-magma-dom/src/components/Button/Button.stories.tsx:142]: Yes please.  Any reference to "foundation" can probably be safely changed to primary-600.

**Review by silvalaura — APPROVED** [review]
Good to go after the build finishes ⭐

**Pull request merged**
